src/Microsoft.ML.StandardLearners/Standard/MultiClass/MultiClassNaiveBayesTrainer.cs

```diff
@@ -170,6 +211,40 @@ private static VersionInfo GetVersionInfo()
 
         public ColumnType OutputType => _outputType;
 
+        /// <summary>
+        /// Copies the label histogram into a buffer.
+        /// </summary>
+        /// <param name="labelHistogram">A possibly reusable array, which will
+        /// be expanded as necessary to accomodate the data.</param>
+        /// <param name="labelCount">Set to the length of the resized array, which is also the number of different labels.</param>
+        public void GetLabelHistogram(ref int[] labelHistogram, out int labelCount)
+        {
+            labelCount = _labelCount;
+            Utils.EnsureSize(ref labelHistogram, _labelCount, _labelCount);
```

Comment:
>_labelCount, _labelCount [](start = 49, length = 24)

Why repeat twice?
Why repeat twice?

We have method which accept only one value, or you deliberately want to shrink array?

Reply:
I was thinking of returning exactly the right size for the array, but I see why if we pass a bigger array it might be better not to shrink it.

---
In reply to: [222538588](https://github.com/dotnet/machinelearning/pull/1111#discussion_r222538588) [](ancestors = 222538588)